crt_portal/api/views.py

```diff
@@ -284,3 +285,32 @@ def get(self, request):
             return HttpResponse(status=400)
         except IndexError:
             return HttpResponse(status=500)
+
+
+class ReferralResponse(APIView):
+    """
+    API endpoint that enables referral letter previews and email send status
+
+
+    Example: api/referral-response/
+    """
+
+    permission_classes = (IsAuthenticated,)
+
+    def post(self, request):
+        report_id = request.data['report_id']
+        report = get_object_or_404(Report, pk=report_id)
+        template_id = request.data['template_id']
+        template = get_object_or_404(ResponseTemplate, pk=template_id)
+        action = request.data['action']
+        preview = {}
+        if action == 'send':
+            try:
+                email_response = crt_send_mail(report, template)
+                if email_response:
+                    response = f'Sent referral email template #{template.id} to report #{report.id}'
+                else:
+                    response = f'Referral email template #{template.id} failed to send to report #{report.id}'
+            except Exception as e:
+                response = f'Referral email template #{template.id} failed to send to report #{report.id}: {e}'
```

Comment:
this try-except is a bit broad (nothing is ever going to go wrong with the `if email_response else` portion unless there's a syntax error in the code, in which case you _don't_ want to catch the exception.

These kinds of nested blocks can be simplified a bit if you're able to return early. For example, maybe instead:

```py
if action == 'send':
  response = {'preview': preview}
  try:
    email_response = crt_send_mail(report, template)
  except Exception as e:
    return Response({'email_response': f'Referral email template #{template.id} failed to send to report #{report.id}: {e}', **response})
  if not email_response:
    return Response({'email_response': f'Referral email template #{template.id} failed to send to report #{report.id}', **response})
  return Response({'email_response': f'Sent referral email template #{template.id} to report #{report.id}', **response})
```

You could further reduce nesting (especially if 'copy' and 'print' turn out to be complicated) by making a helper function for the `if action == 'send'` portion